workflow/controller/operator.go

```diff
@@ -954,6 +954,13 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 			if node.Succeeded() && match {
 				woc.completedPods[pod.Name] = pod.Status.Phase
 			}
+			if node.Type == wfv1.NodeTypePod {
```

Comment:
think we might want this earlier in this func, e.g. before line 945 which is a return